views/js/qtiCreator/widgets/interactions/uploadInteraction/states/Question.js

```diff
@@ -39,25 +41,35 @@ define([
         // Pre-select a value in the types combo box if needed.
         let preselected = uploadHelper.getExpectedTypes(interaction);
 
+        types.unshift({ mime: ANY_KIND, label: __('-- Any kind of file --') });
+
+        if (interaction.attr('type') === '' || preselected.includes(ANY_KIND)) {
+            // Kill the attribute if it is empty or has any/kind
+            interaction.removeAttr('type');
+            interaction.removeAttr('class');
+            preselected = [];
+        }
+
         const config = module.config();
 
         if (preselected.length === 0 && config.defaultList && config.defaultList.length > 0) {
             preselected = preselected.concat(config.defaultList);
             uploadHelper.setExpectedTypes(interaction, config.defaultList);
         }
 
-        types.unshift({ "mime" : "any/kind", "label" : __("-- Any kind of file --") });
-
-        if (interaction.attr('type') === '') {
-            // Kill the attribute if it is empty.
-            delete interaction.attributes.type;
-        }
-
-        for (let i in types) {
-            if (_.indexOf(preselected, types[i].mime) >= 0) {
-                types[i].selected = true;
+        if (preselected.length === 0) {
+            // WHEN loading the item in Authoring
+            // AND (the file types contains only -- Any kind of file -- OR the file types contains no type at all)
+            // THEN only -- Any kind of file -- is shown in the selection box.
+            types[0].selected = true;
+        } else {
+            for (let i in types) {
```

Comment:
**quibble:** Better to use `forEach()` instead of a `for..in` loop over an array, it is less error-prone. If it is for performance purposes, then the fastest would be the classical `for(;;)` with a cached copy of the array length.